x/stakeibc/keeper/reward_converter.go

```diff
@@ -133,57 +133,56 @@ func (k Keeper) TransferConvertedTokensTradeToHost(ctx sdk.Context, amount sdk.I
 // The amount represents the total amount of the reward token in the trade ICA found by the calling ICQ
 // Depending on min and max swap amounts set in the route, it is possible not the full amount given will swap
-func (k Keeper) TradeRewardTokens(ctx sdk.Context, amount sdk.Int, route types.TradeRoute) error {
+func (k Keeper) TradeRewardTokens(ctx sdk.Context, rewardAmount sdk.Int, route types.TradeRoute) error {
 	// If the min swap amount was not set it would be ZeroInt, if positive we need to compare to the amount given
 	//  then if the min swap amount is greater than the current amount, do nothing this epoch to avoid small swaps
-	if route.MinSwapAmount.IsPositive() && route.MinSwapAmount.GT(amount) {
+	if route.MinSwapAmount.IsPositive() && route.MinSwapAmount.GT(rewardAmount) {
 		return nil
 	}
 
 	// If the max swap amount was not set it would be ZeroInt, if positive we need to compare to the amount given
 	//  then if max swap amount is LTE to amount full swap is possible so amount is fine, otherwise set amount to max
-	if route.MaxSwapAmount.IsPositive() && route.MaxSwapAmount.GT(amount) {
-		amount = route.MaxSwapAmount
+	if route.MaxSwapAmount.IsPositive() && route.MaxSwapAmount.GT(rewardAmount) {
+		rewardAmount = route.MaxSwapAmount
 	}
 
-	// See if pool swap spot price has been set to a valid ratio (string representing a float like "10.203")
-	// If there is a valid spot price, use it to set a floor for the acceptable minimum output tokens
-	// 5% slippage is allowed so multiply by 0.95 the expected spot price to get the target minimum
+	// See if pool swap price has been set to a valid ratio
+	// If there is a valid price, use it to set a floor for the acceptable minimum output tokens
 	// minOut is the minimum number of route.TargetDenomOnTradeZone we must receive or the swap will fail
-	minOut := sdk.ZeroInt()
-	if route.SpotPrice != "" {
-		if spotPrice, err := sdk.NewDecFromStr(route.SpotPrice); err == nil {
-			slippageRatio := sdk.NewDecWithPrec(95, 2) // 0.95 to allow 5% loss to slippage
-			inputToOutputRatio := slippageRatio.Mul(spotPrice)
-			minOut = sdk.NewDecFromInt(amount).Mul(inputToOutputRatio).TruncateInt()
-		} else {
-			k.Logger(ctx).Error("Couldn't parse the spot price %s as a Decimal", route.SpotPrice)
-			// Don't allow a missing spot price to stop the swap, use zeroInt as the lower bound for now
-		}
-	}
+	//
+	// To calculate minOut, we first convert the rewardAmount into units of HostDenom,
+	//   and then we multiply by (1 - maxSlippage)
+	//
+	// The price of the host token is denominated in units of reward tokens
+	// Meaning, a price of 1.1 implies 1.1 RewardTokens are needed to buy 1 HostToken
+	// So, to convert from units of RewardTokens to units of HostTokens,
+	// we divide the reward amount by the price:
+	//   AmountInHost = AmountInReward / PriceOfHostRelativeToReward
+	if route.HostTokenPrice == sdk.ZeroDec() {
```

Comment:
Careful on this check, in the code below for TradeRoute initialization you use sdk.OneDec to be the starting value, maybe we should use something literally impossible like a negative number sdk.Dec to explicitly mean not initialized since zero could be a legal value still... although we probably wouldn't want to do that trade if it were zero

Reply:
Sorry I see now that was just initialization for a test not for all traderoute setups.  Still we might want an impossible value to start out

Reply:
I think zero should be impossible unless the pool is drained, in which case I think we'd also want to reject the trade

I'd also be fine with -1, it's just a bit clunkier with sdkmath (I think you have to do `sdk.OneDec().MulInt64(-1)`)